### DISCUSSION TITLE
Why would I want to copy a string defined in source into a file specified by target?

### DISCUSSION THREAD

**Discussion by jgarte** [useblocks/sphinx-collections#11] [Q&A]
Hi,

Why would I want to copy a string defined in source into a file specified by target?

https://sphinx-collections.readthedocs.io/en/latest/drivers/string.html

What is an example of when this would be needed in a monorepo project? 

What particular problem/user story did it solve for the software development team inside the german automotive company?

**Comment by danwos** [comment]
All the Sphinx-Collections features are not necessarily related to the monorepo use case.
"monorepo" does not tell you anything about how the documentation is structured and how it needs to get built.
Please see [my view on monorepo in this comment](https://github.com/useblocks/sphinx-collections/issues/10#issuecomment-1406123088).

However, use cases for getting (dynamic) strings into a project are:
- Add custom names for custom builds. For instance, create a customer-specific html-build, with the customer name in it.
- Provide a landing page file, which toctree got calculated (Generate the string inside a function and then copy it)
- Add a changelog based on the latest git-commits (own function provides the string)
- Update dates, contact information or whatever and add them to a file (e.g. LICENSE.txt with always update year and contact person)

But as you see, these use cases are not really monorepo specific, but could be useful there as well :)
